.github/workflows/reusable-python-publish_package.yml

```diff
@@ -0,0 +1,176 @@
+            CONDA_CHANNEL_PRIORITY: strict
+            PIP_NO_DEPS: 1 # all dependencies are installed from conda
+        timeout-minutes: 20
+        if: ${{ inputs.package_manager == 'conda' }}
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v2
+                with:
+                    lfs: ${{ inputs.lfs }}
+            -   name: Check if the version tag is available
+                run: |
+                    if [ "${{ inputs.version_tag }}" = "v0.0.0" ]; then
+                        echo "Version tag not available"
+                        exit 1
+                    else
+                        echo "Version tag available"
+                    fi
+            -   name: Check if there is a conda recipe
+                run: |
+                    if [ -f ${{ inputs.package_name }}/meta.yaml ]; then
+                        echo "Conda recipe found"
+                    else
+                        echo "Conda recipe not found"
+                        exit 1
+                    fi
+            -   name: Setup CONDA_LOCK_ENV_FILE
+                run: |
+                    os_conda=$( [[ $RUNNER_OS == "Linux" ]] && echo "linux" || ([[ $RUNNER_OS == "Windows" ]] && echo "win" || echo "osx"))
+                    
+                    echo "CONDA_LOCK_ENV_FILE=environments/py-${{ inputs.python_version }}-${os_conda}-64-dev.conda.lock.yml" >> $GITHUB_ENV
+            -   uses: MiraGeoscience/CI-tools/.github/actions/reusable-python-setup_conda@main
+                name: Setup conda env
+                with:
+                  python_ver: ${{ inputs.python_version }}
+            -   name: Install dependencies
+                run: conda install -y conda-build conda-verify
+            -   name: Build package
+                id: build_package
+                run: |
+                    conda build ${{ inputs.package_name }}
+                    build_path=$(conda build --output ${{ inputs.package_name }})
+                    build_name=$(basename ${build_path})
+                    echo "BUILD_PATH=${build_path}" >> $GITHUB_ENV
+                    echo "BUILD_NAME=${build_name}" >> $GITHUB_ENV
+            -   name: Publish
+                run: |
+                    MIRA_REPO_CONFIG=$(echo ${{ secrets.MIRA_REPO_SECRETS }} | base64 -d)
+                    repository_list=$(echo '${{ inputs.repo_names }}' | jq -r '.[]')
+                    for repo_name in ${repository_list}; do
+                        repo_token=$(echo ${MIRA_REPO_CONFIG} | jq -r --arg repo_name "$repo_name" '.repositories[] | select(.name == $repo_name) | .token')
+                        repo_url=$(echo ${MIRA_REPO_CONFIG} | jq -r --arg repo_name "$repo_name" '.repositories[] | select(.name == $repo_name) | .url')
+                        curl -T ${{ env.BUILD_PATH }} -H "Authorization: Bearer ${repo_token}" "${repo_url}/${{ inputs.package_name }}/${{ inputs.version_tag }}/${{ env.BUILD_NAME }}"
```

Comment:
**question:** is repo token sent in clear over the network. Might be more secure with a jfrog login.

See the github action: https://github.com/jfrog/setup-jfrog-cli
and doc about the JFrog CLI:
- for authentication: https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli/cli-for-jfrog-artifactory/authentication
- for uploading file: https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli/cli-for-jfrog-artifactory/generic-files
- curl integration for generic purpose: https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli/cli-for-jfrog-artifactory/curl-integration

I have also found `--oauth2-bearer <token>` since curl 7.61.0 (no sure if more secured than header...)